hilti/toolchain/src/compiler/codegen/codegen.cc

```diff
@@ -703,7 +703,7 @@ cxx::declaration::Function CodeGen::compile(const ID& id, type::Function* ft, de
     if ( linkage == declaration::Linkage::Struct ) {
         // For method implementations, check if the ID is fully scoped with
         // the module name; if so, remove.
-        if ( id.sub(0) == _hilti_module->uid() )
+        if ( id.sub(0).str() == _hilti_module->uid().str() )
```

Comment:
We currently compute the module `UID::str` twice which allocates a `std::string` each time. Could you either change the return type of `UID::str` to `const std::string&`, or alternatively cache the value?